### DISCUSSION TITLE
Can't add nodes

### DISCUSSION THREAD

**Discussion by leftyfb** [gravitational/teleport#5700] [Q&A] [answered]
I can't seem to add any nodes:

sudo teleport start --roles=node --auth-server=https://hostname:3080 --token=XXX --labels=env=lxc
ERRO [PROC:1] Node failed to establish connection to cluster: ssh: handshake failed: no matching keys found. time/sleep.go:148

The server says:
Feb 24 21:47:00 teleport teleport[1235]: 2021/02/24 21:47:00 http: TLS handshake error from ip.address:45748: remote error: tls: bad certificate

What am I doing wrong?

**Comment by leftyfb** [comment]
ok, I found part of the problem. The node I was trying was an lxd container behind it's NAT'd interface. Apparently that doesn't work without some additional configuration. I'm now trying a regular linux device which isn't on the same network as the server. Everything seems to go ok with teleport finally running on the node and showing up in the list of nodes, but trying to connect I get:

failed connecting to node <HOSTNAME>. dialing through a tunnel: no tunnel connection found: no node reverse tunnel for 182ecb6c-2732-40c1-86ba-44a05b8ebb6c.teleport found, dialing directly: dial tcp <IP.ADDRESS>:3022: i/o timeout

**Reply by leftyfb** [reply]
alright, I guess there's some conflicting documentation where you use different ports for different reasons. I am now able to get a regular node to connect but still not the lxd container

**Reply by webvictim** [reply]
Just for a little clarification:

- If you join a node to port 3080 (i.e. via the Teleport proxy) then the node doesn't listen on port 3022; all communication has to be tunnelled via the proxy.
- If you join a node to port 3025 (i.e. via the Teleport auth server) then the node listens on port 3022 and connections can be made that way.

**Comment by leftyfb** [comment, marked as answer]
Got help from Slack.
Just needed to delete /var/lib/teleport on the node
